am-core-web-service/src/main/java/org/am/core/web/service/admingeneral/ClassroomService.java

```diff
@@ -0,0 +1,113 @@
+package org.am.core.web.service.admingeneral;
+
+import org.am.core.web.domain.entity.admingeneral.Classroom;
+import org.am.core.web.dto.admingeneral.ClassroomDto;
+import org.am.core.web.dto.admingeneral.ClassroomRequest;
+import org.am.core.web.repository.jpa.CustomMap;
+import org.am.core.web.repository.jpa.admingeneral.ClassroomRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class ClassroomService implements CustomMap<ClassroomDto, Classroom> {
+    private final ClassroomRepository classroomRepository;
+
+    public ClassroomService(ClassroomRepository classroomRepository) {
+        this.classroomRepository = classroomRepository;
+    }
+
+    public List<ClassroomDto> getActiveClassroomByAreaId(Integer areaId){
+        List<Classroom> classrooms = classroomRepository.findAllByArea_IdOrderById(areaId);
+        List<Classroom> activeClassrooms = classrooms.stream()
+                .filter(c->c.getActive().equals(Boolean.TRUE))
+                .collect(Collectors.toList());
+        return activeClassrooms.stream()
+                .map(this::toDto)
+                .collect(Collectors.toList());
+    }
+
+    public Optional<ClassroomDto> getClassroomByIdByArea(Integer id){
+        Classroom classroom = classroomRepository.findById(id).orElseThrow();
+        Integer areaId = classroom.getArea().getId();
+
+        List<ClassroomDto> classrooms = getActiveClassroomByAreaId(areaId);
+
+        ClassroomDto foundClassroom = classrooms.stream()
+                .filter(c-> c.id().equals(id))
+                .findFirst()
+                .orElseThrow();
```

Comment:
You have to change the implementation getById ... you just need id, check itineraryService or AreaService